### ISSUE TITLE
Connection refused from container to container communication

### ISSUE THREAD

**Issue by thyeestes** [traefik/traefik#10910]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you do?

I upgraded my traefik container to v3.1 to enable socket activation. I removed PublishPort from my container file and created two socket files( one for HTTP and one for HTTPS). When my Traefik container is started I can acces my containers from my LAN, but Uptime Kuma cannot reach the websites proxied by Traefik. 
I also tested with a container running a browser but got the same result. All containers are on the same network. 

Turning off socket activation solves the issue.

### What did you see instead?

Kuma reports an error: connect ECONNREFUSED 192.168.1.100:443

I also notice that there are more processes listening on port 80 and 443, so maybe over the Podman network a connection is made to the wrong listener, i.e. not traefik port 443

Ports listening without socket activation:
rootlessp 589917 thyestes   11u  IPv6 2247907      0t0  TCP *:80 (LISTEN)
rootlessp 589917 thyestes   12u  IPv6 2247908      0t0  TCP *:443 (LISTEN)

Ports listening with socket activation:
systemd      775 thyestes   34u  IPv4 2564674      0t0  TCP *:80 (LISTEN)
systemd      775 thyestes   35u  IPv4 2566204      0t0  TCP *:443 (LISTEN)
conmon    666883 thyestes    3u  IPv4 2566204      0t0  TCP *:443 (LISTEN)
conmon    666883 thyestes    4u  IPv4 2564674      0t0  TCP *:80 (LISTEN)
traefik   666896 thyestes    3u  IPv4 2564674      0t0  TCP *:80 (LISTEN)
traefik   666896 thyestes    5u  IPv4 2566204      0t0  TCP *:443 (LISTEN)

### What version of Traefik are you using?

Version:      3.1.0
Codename:     comte
Go version:   go1.22.5
Built:        2024-07-15T14:44:04Z
OS/Arch:      linux/amd64

### What is your environment & configuration?

```yaml
[Container]
Image=docker.io/library/traefik:v3.1
ContainerName=traefik
Network=proxy.network
Volume=/storage/traefik/acme.json:/acme.json
Volume=%t/podman/podman.sock:/var/run/docker.sock
PublishPort=80:80
PublishPort=443:443
```

Add more configuration information here.
Client:       Podman Engine
Version:      5.1.2
API Version:  5.1.2
Go Version:   go1.21.12
Built:        Thu Jul 11 15:01:37 2024
OS/Arch:      linux/amd64

**Comment by eriksjolund** [comment]
When using socket activation of containers with Podman, there is no need to specify

PublishPort=80:80
PublishPort=443:443

Have you tried without those lines?

**Comment by thyeestes** [comment]
Yes I have. This was the file without socket activation. Updated my initial post to comment out those two lines. This is the quadlet file when I have the two socket files enabled and started.

**Comment by eriksjolund** [comment]
I assume you are using the network software pasta. One idea: ask in the mailing list passt-user@passt.top
(see https://passt.top/passt/lists)

I think the problem described 
in this issue is analogous to this problem:

Create an outbound connection from a container running in a custom network to a service running on the host.

I'm not sure it is currently possible with pasta.

**Comment by eriksjolund** [comment]
I see in the error message that the Uptime Kuma container tried to connect to 192.168.1.100.
Is that the IP address of the Trafik container on the custom network (that is defined in the file proxy.network)?

If so, maybe it's possible to configure the Uptime Kuma container to use an external DNS server so that the IP address is resolved to the IP address of the host?

**Comment by thyeestes** [comment]
192.168.1.100 is the IP address of my host, so DNS is working fine. All my containers are configured in the proxy network, 192.168.60.x.

**Comment by eriksjolund** [comment]
> One idea: ask in the mailing list passt-user@passt.top
> (see https://passt.top/passt/lists)

Probably we need some help from the pasta developers. Another way to reach them is
by creating a new Podman GitHub discussion thread
https://github.com/containers/podman/discussions

**Comment by thyeestes** [comment]
Thanks I will start a discussion in the Podman repository, will be tomorrow though.

I have fairly simple setup to reproduce:
- Podman running on a Debian host
- Two containers: Traefik and Kuma both connected to the same podman network
- From Kuma I monitor the Traefik dashboard

**Comment by rtribotte** [comment]
Hello @thyeestes @eriksjolund,

Thanks for your interest in Traefik,

This issue seems to be Podman configuration-related only and not a bug in Traefik.

However, the GitHub issue tracker is dedicated to bug and feature requests.
For help on your configuration, please join our [Community Forum](https://community.traefik.io/) and reach out to us on the [Traefik section](https://community.traefik.io/c/traefik). 

We will close this issue accordingly but feel free to re-open it if you think we missed something.

**Comment by thyeestes** [comment]
Okay understood and thanks
